core/vm/contracts.go

```diff
@@ -109,6 +112,24 @@ var PrecompiledContractsBLS = map[common.Address]PrecompiledContract{
 	common.BytesToAddress([]byte{18}): &bls12381MapG2{},
 }
 
+// PrecompiledContractsHilbert contains the default set of pre-compiled Ethereum
+// contracts used in the Hilbert release.
+var PrecompiledContractsHilbert = map[common.Address]PrecompiledContract{
+	common.BytesToAddress([]byte{1}): &ecrecover{},
+	common.BytesToAddress([]byte{2}): &sha256hash{},
+	common.BytesToAddress([]byte{3}): &ripemd160hash{},
+	common.BytesToAddress([]byte{4}): &dataCopy{},
+	common.BytesToAddress([]byte{5}): &bigModExp{eip2565: true},
+	common.BytesToAddress([]byte{6}): &bn256AddIstanbul{},
+	common.BytesToAddress([]byte{7}): &bn256ScalarMulIstanbul{},
+	common.BytesToAddress([]byte{8}): &bn256PairingIstanbul{},
+	common.BytesToAddress([]byte{9}): &blake2F{},
+
```

Comment:
`common.BytesToAddress([]byte{5}): &bigModExp{eip2565: true}` ==> `	common.BytesToAddress([]byte{5}):&bigModExp{},`

and add 
```
common.BytesToAddress([]byte{100}): &tmHeaderValidate{},
common.BytesToAddress([]byte{101}): &iavlMerkleProofValidate{},
```

Reply:
sorry, even you add `PrecompiledContractsHilbert` here, but seems like you did not use it, can you refer to how we use `PrecompiledContractsIstanbul`. Otherwise, the newly added native contracts won't take effect.

Reply:
I have added changes to include Hilbert precompiles, I tested by running an actual solidity test. however, I'm unsure if I did make all the necessary changes there are a few things that I did not understand when I referred to Istanbul such as the Instruction set/ tx pool changes that Istanbul has, I assume it's irrelevant for this case, and ignored it.
Also, I set `nil` to the block numbers in all the chain config. assuming you will change it when you move it to testnet / mainnet
Please do let me know if I had missed anything. Thank you.